oauth2client/contrib/gce.py

```diff
@@ -85,32 +142,126 @@ class AppAssertionCredentials(AssertionCredentials):
     information to generate and refresh its own access tokens.
     """
 
-    @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    def __init__(self,
+                 scope=None,
+                 service_account_email='default',
+                 service_account_info=None,
+                 **unused_kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. The default service account is
+                usually the shared Compute Engine service account.
+            service_account_info:
+                (Optional) Dictionary containing information about a
+                service account (typically deserialized from JSON).
+                If passed in, will be used as ``service_account_info``
+                property. Otherwise, the Compute Engine metadata server
+                will be used to determine this value. Additionally,
+                if present the 'email' field in this argument must be
+                provided and will override the ``service_account_email``
+                argument.
         """
-        if scope:
-            warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
-        self.kwargs = kwargs
-
         # Assertion type is no longer used, but still in the
         # parent class signature.
-        super(AppAssertionCredentials, self).__init__(None)
-        self._service_account_email = None
+        super(AppAssertionCredentials, self).__init__(
+            None,
+            scopes=scope,
```

Comment:
`scope` is definitely misnamed. I'd prefer to remove it all together. See below.

It's my habit to call `unused_kwargs` all those kwargs which are passed to the parent constructor. 

However. `AssertionCredentials` doesn't do that. Which is actually a big mistake because it prevents me from overriding the scopes usage in the parent `GoogleCredentials` class. So these are actually even more unused than my normal usage of `unused_kwargs`. I can change back to  `kwargs` if it bothers you though.